src/components/basic_table/custom_item_action.tsx

```diff
@@ -1,12 +1,12 @@
 import React, { Component, cloneElement } from 'react';
 import { CustomItemAction as Action } from './action_types';
-import { Item, ItemId } from './table_types';
+import { ItemId } from './table_types';
 
-interface Props {
-  action: Action;
+export interface Props<T> {
```

Comment:
I think we'd want the naming to be more specific than `Props`? Makes it easier to track in consuming files. (here & in the other files)